Clarifie l'estimation du temps de lecture

mehdbnaleg opened the pull request:
Tente de résoudre l'issue #5780.

# Q/A

1. Prendre l'archive du tutoriel Arduino
2. Le publier
3. Regarder l'estimation du temps de lecture
4. Constatez que c'est 1 jour 6h

# Comportement attendu

L'estimation est de 30h.

PS : Le template fonctionne : l'erreur est dûe à un comportement bizarre qui affecte probablement les dernières lignes de `tags_authors.part.html`. Voir l'issue #5995 pour plus de détails.

Merci !

Situphen committed:
Met à jour factory-boy

AmauryCarrade committed:
Import des normes depuis #5835

AmauryCarrade committed:
Ajustement de la palette

Ajout d'une couleur supplémentaire par catégorie et augmentation de la
luminosité des bleus des premières teintes primaires.

AmauryCarrade committed:
Première vague de normalisation du CSS

AmauryCarrade committed:
Seconde vague de normalisation (normalement c'est terminé)

Avec quelques retouchessur la première vague au passage, et une légère
altération de la palette au niveau des oranges (couleurs d'accentuation).

Il ne manque que la documentation à récupérer de l'autre PR et à compléter.

AmauryCarrade committed:
Bug dans le SCSS quand construit via make build-front

AmauryCarrade committed:
Retouches diverses

- Reprise de l'affichage des membres, en individuel et en liste
- Reprise de la zone métadonnée des contenus
- Autres détails ça et là

AmauryCarrade committed:
Reprise et finition de la documentation de #5835

La documentation commencée dans la PR #5835 a été récupérée et augmentée
avec toutes les normes déclarées et quelques conseils afin que le code
SCSS du site reste propre.

AmauryCarrade committed:
Retouche des listes d'utilisateurs.

Au passage, retouche de la page des membres par IP pour coller au CSS.

AmauryCarrade committed:
Nouvelle mise en forme des messages (bureau)

Reste à faire la version mobile

AmauryCarrade committed:
Accessibilité des nouveaux messages

La navigation au clavier est supportée sur le menu déroulant.

AmauryCarrade committed:
Corrige l'affichage des profils vides.

Le bloc bio doit prendre toute la largeur ; or ce n'était pas le cas.

Situphen committed:
Mise à jour de python-slugify et django-uuslug

Situphen committed:
Spécifie la valeur de force_slug_update lors des content.save()

mehdbnaleg committed:
Clarifie l'estimation du temps de lecture (non fonctionel)

mehdbnaleg committed:
Change les & en "and"

mehdbnaleg committed:
Meilleure gestion du pluriel

mehdbnaleg committed:
Meilleure gestion du pluriel

mehdbnaleg committed:
Modifie les test suite aux modifications du template

mehdbnaleg committed:
Modifie les test suite aux modifications du template

AmauryCarrade committed:
Amélioration des fiches de contenus

- Fiches moins denses sur l'accueil avec le retrait des catégories
  (dont l'intérêt laisse à désirer sur l'accueil).
- Reprise du CSS afin de mieux gérer le positionnement des différents
  éléments (en particulier les métadonnées).
- Correction d'un bug d'affichage où les appareils avec une haute
  densité de pixels voyaient une image de fond sans aucun rapport.

AmauryCarrade committed:
Retour du soulignement au survol des fiches de contenus.

AmauryCarrade committed:
Amélioration (tentative) de la zone de nouveau tuto sur l'accueil

- Le texte n'est plus “délavé”.
- Nouvel effet au survol, plus marquant qu'un changement de
  0,2% de teinte.

AmauryCarrade committed:
Pied de page manquant de marges

AmauryCarrade committed:
Correction du titre des modales par dessus les icônes

AmauryCarrade committed:
Correction de l'alignement des icônes des titres

AmauryCarrade committed:
Retrait d'un zindex inutile

Qui en plus faisait passer le texte au dessus de certains menus

AmauryCarrade committed:
Correction de marges

AmauryCarrade committed:
Mauvais affichage de l'en-tête des contenus sur mobile

mehdbnaleg committed:
Retire les imports inutiles + code plus propre

mehdbnaleg committed:
Retire les imports inutiles + code plus propre

mehdbnaleg committed:
Diminue la longueur de ligne + corrige erreur d'import

mehdbnaleg committed:
Diminue la longueur de ligne + corrige erreur d'import

mehdbnaleg committed:
Rend le code plus lisibe (correct cette fois)

mehdbnaleg committed:
Rend le code plus lisibe (correct cette fois)

mehdbnaleg committed:
Ameliorations stylistiques mineures du code + import ngettex

mehdbnaleg committed:
Ameliorations stylistiques mineures du code + import ngettex

mehdbnaleg committed:
Modifications de style (PEP-8)

Situphen committed:
Par défaut, ne met pas à jour le slug lors des content.save() (force_slug_update=False)

Situphen committed:
Correction d'un test

AmauryCarrade committed:
Merge branch 'dev' into issue/5780

Arnaud-D committed:
Utiliser black pour formater le code Python (#5964)

* Reformate le code Python avec black

* Inclut les migrations

* Add black to CI instead of flake8

* Reformat new files since rebase

Situphen committed:
Ajout d'un wrapper autour de django-uuslug et renommage des fonctions slugify()

Situphen committed:
Mise à jour triviale des paquets Node JS (#6005)

Situphen committed:
Mise à jour triviale des dépendances Python (#6004)

Co-authored-by: Amaury Carrade <amaury@carrade.eu>

Situphen committed:
Permet de faire tomber des cœurs pour la Saint-Valentin (merci à Sandhose) (#6002)

Situphen commented:
@ToxicScorpius Il y a des conflits à résoudre par rapport à la branche `upstream/dev` étant donné que la PR qui met en place Black a été fusionnée et à modifié le style des fichiers Python. :)

firm1 committed:
suppression de jquery du fichier close-alert-box.js (#5969)

* migrate to vanilla js close alert box.js

* remove setTimeout

Co-authored-by: Situphen <Situphen@users.noreply.github.com>

AmauryCarrade committed:
L'installeur en version complète ne persiste plus définitivement le PATH (#5944)

L'installation du JDK local (pour ElasticSearch) avait un bug extrêmement
fourbe : suite à une petite erreur dans le script d'installation, le
script d'activation de l'environnement virtuel persistait définitivement
le PATH du moment de l'installation dans l'environnement virtuel. Une des
conséquences est de fixer la version de Node dans l'environnement virtuel
à celle lors de l'installation, car cette modification du PATH est placée
après `nvm use` dans le fichier `zdsenv/bin/activate` (et assimilés),
provoquant des situations assez incompréhensibles où à l'activation du
venv, nvm précise qu'il passe dans la bonne version de Node, mais une autre
se retrouve active à la fin de l'exécution de la commande.

Ce commit corrige ce comportement en évitant de persister le PATH du moment
de l'installation dans le fichier `zdsenv/bin/activate` (ou assimilé).

Si vous avez ce problème, le moyen le plus simple de le résoudre est de
supprimer ces trois lignes du fichier `zdsenv/bin/activate`, placées à la
toute fin du fichier, normalement :

export PATH="..."
export JAVA_HOME="..."
export ES_JAVA_OPTS="..."

…puis d'exécuter une toute petite partie du script d'installation afin de les
restaurer correctement (ce qui ne devrait pas prendre plus de quelques secondes,
dépendant de votre connexion internet).

./scripts/install_zsh.sh +jdk-local

Co-authored-by: Situphen <Situphen@users.noreply.github.com>

Arnaud-D committed:
Découple le module de forum du module de notification (#5976)

* Topic moved

* Topic edited

* Topic read and post read

* Post unread

* Fix argument name

* Try again!

* Fix shadowed function

* Fix self signaling in notifications

* Et de un

* Et de deux

* Lint

Co-authored-by: Situphen <Situphen@users.noreply.github.com>

Arnaud-D committed:
Découple le module de notification et le module de tuto (#5982)

* Découple le module de notification et le module de tuto

* Reformat new files since rebase

* Corrige des erreurs de merge

Arnaud-D committed:
Supprime le signal inutilisé `answer_unread`

Arnaud-D committed:
Supprime le signal inutilisé `edit_content`

AmauryCarrade committed:
Modification des noirs et de la couleur des contenus

- Le noir impur est à peine plus clair.
- Les contenus sont affichés en noir et non en gris sombre, améliorant
  ainsi le confort de lecture.

AmauryCarrade committed:
Fermeture de la bannière synchronisée sur tous les onglets si définitif (#5945)

Si la bannière est fermée avec Ctrl+clic ou Maj+clic, elle est fermée sur
tous les onglets actifs instantanément.

AmauryCarrade committed:
Ajustement des couleurs de l'en-tête

- La couleur du survol des liens et du contenu des blocs
  déroulés est plus claire.
- La couleur au survol des liens dans les blocs déroulés
  a été ajustée afin que le contraste soit moins violent.

AmauryCarrade committed:
Correction de la bordure de l'incitation de rédaction.

- Elle était grise. Elle est bleue. Parfois, bordure
  évolue.

AmauryCarrade committed:
Tous les pieds de contenus sur l'accueil sont orange.

- Même les sujets.
- Fin de la différenciation.
- Tout le monde à la même enseigne.

AmauryCarrade committed:
Contenus sur l'accueil : seul le titre est souligné

- Le sous-titre ne l'est plus.
- C'est moins trompeur.
- Le côté cliquable est toujours mis en avant par le
  soulignement du titre.
- Moins de soulignement c'est toujours une bonne nouvelle.

AmauryCarrade committed:
Attention ça pique.

- Les arrondis sont pointus.
- Cohérence avec le reste.
- Ça reviendra peut-être.
- Attention aux doigts.

AmauryCarrade committed:
La bulle de commentaires est sauvée.

- Fin du harcèlement de la bulle par la liste des
  auteur⋅e⋅s.
- Elle était écrasée, elle ne l'est plus.
- S'il y a trop d'auteur⋅e⋅s (je vous vois, les récaps),
  iels s'abbrègeront.

AmauryCarrade committed:
Un fond moins sombre pour les modales.

- Le fond des modales fait polémique.
- On tente des choses.
- Expérimental, pourra toujours évoluer.

AmauryCarrade committed:
Correction du bouton d'ajout d'un membre à un MP.

- Lui aussi tentait l'art moderne.
- Là non plus, pas un succès.
- Il est plus sage.

Arnaud-D committed:
Déplace les signaux de ping et unping dans le module `utils`

Arnaud-D committed:
Renommage des récepteurs pour les signaux `ping` et `unping`

AmauryCarrade committed:
Meilleur affichage des bulles.

- Les bulles se montrent dans toute leur splendeur.
- Leur pointe n'est plus injustement censurée.
- Même le fond blanc ne leur fait pas peur et ne les
  masque plus. #BlackLivesMatter
- L'ombre est la première standard, mais assombrie pour
  compenser le fait qu'on ne peut en mettre deux.

Situphen committed:
Utilise la fonction django.utils.html.escape() pour échapper les apostrophes dans les tests

Situphen committed:
Renomme `from django.template.base import Context` en `from django.template import Context`

Situphen committed:
Renomme force_text() en force_str()

Situphen committed:
Renomme smart_text() en smart_str()

Situphen committed:
Renomme `django.utils.http.urlquote()` en `urllib.parse.quote()`

Situphen committed:
Renomme `django.utils.http.urlunquote()` en `urllib.parse.unquote()`

Situphen committed:
Renomme les fonctions `ugettext()` en `gettext()`

AmauryCarrade committed:
Léger ajustement des couleurs primaires

- La couleur primaire 400 est un peu moins “flashy” (bon, à peine,
  hein).

AmauryCarrade committed:
Amélioration du CSS des boîtes de contenu.

- Variabilisation des parties communes et nettoyage.
- Les billets ont des dates.

AmauryCarrade committed:
Meilleure lisibilité des contenus.

- La hauteur des lignes a été augmentée à 1.6.

AmauryCarrade committed:
Ajustement des bulles d'informations aux nouvelles couleurs

AmauryCarrade committed:
Ajustement des coches et des pouces aux nouvelles couleurs

AmauryCarrade committed:
Retour des marges entre blocs sur les grands écrans

- Une règle sur les grands écrans avait des effets de bord en
  supprimant beaucoup trop de marges de façon très violente
  (ce !important du démon…).

AmauryCarrade committed:
Titres plus compacts

- Les titres des contenus étaient démesurément espacés, dés qu'ils
  passaient sur plusieurs lignes.

AmauryCarrade committed:
Meilleures marges dans les grands titres de contenus

AmauryCarrade committed:
Correction du sommaire des tutoriels

AmauryCarrade committed:
Reprise des alertes

- Plus lisibles
- Plus jolies
- Découplées des bandeaux de notification !

AmauryCarrade committed:
Correction de l'affichage de la recherche.

- Les résultats de la recherche n'avaient pas été convertis au
  nouveau format des cartes.

AmauryCarrade committed:
Correction des marges entre blocs sur l'accueil

AmauryCarrade committed:
Correction de dépassements dans les extraits de forum

- Dépassements sur la page d'accueil et les pages de profil sur les
  extraits de forum, sur les très petits écrans

AmauryCarrade committed:
Diminution de l'interligne de la citation de l'accueil

AmauryCarrade committed:
Toutes les alertes utilisent le même modèle graphique.

- Déplacement de la logique d'affichage des alertes dans un unique
  gabari.
- Ce gabari permet de spécifier l'URL à utiliser pour résoudre.
- Les contenus, les profils, et les messages, utilisent désormais
  ce gabari.

AmauryCarrade committed:
Correction des marges du bandeau de pagination.

- Lorsqu'un contenu ne se termine pas par un paragraphe (par exemple,
  par une vidéo), le bandeau de pagination n'est plus collé au
  contenu.

AmauryCarrade committed:
La page n'est plus trouée.

- Retrait de la marge au dessus du pied de page, qui n'était pas
  vraiment utile, et qui faisait apparaître, sur mobile, le menu
  latéral caché sous le conteneur principal (`.main-container`).
- Correction de marges incorrectes des alertes sur mobile.

AmauryCarrade committed:
Les notifications étaient mal alignées

- Suite à un effet de bord, les éléments de la liste des notifications
  (au sens large : alertes, MPs ou notifications) étaient mal alignées.
- Le bloc flottant des notifications était mal positionné sur tablette.

AmauryCarrade committed:
Correction de l'alignement de l'en-tête sur mobile

AmauryCarrade committed:
Les blocs d'information sont plus clairs.

En conséquence les mini-codes dans ces blocs sont plus sombres.

AmauryCarrade committed:
Correction de l'alignement des titres des modales sur mobile

AmauryCarrade committed:
Correction de l'affichage de la liste des participants d'un MP

- Aussi, corrige l'affichage de l'auteur d'un sujet du forum

AmauryCarrade committed:
Correction de la position des menus de notifications

AmauryCarrade committed:
Correction de l'affichage de la liste des contributeurs

L'objectif initial était de corriger un effet de survol incorrect dans
feu le tableau, mais j'ai pris la liberté d'utiliser le nouveau
composant de liste d'utilisateurs, ce qui permet un meilleur rendu
sans passer par un artificiel tableau.

AmauryCarrade committed:
Correction des marges dans les boîtes modales

- Correction des marges générales, qui n'étaient pas très cohérentes
- Correction des marges dans le cas spécifique de formulaires qui ont
  une marge séparatrice superflue dans certains cas.

AmauryCarrade committed:
Support pour les listes de membres en pleine largeur.

AmauryCarrade committed:
Ajout de la documentation pour les listes de membres.

AmauryCarrade committed:
`.authors` → `.members`

C'est beaucoup plus logique vu que c'est utilisé pour plein d'autres
choses.

AmauryCarrade committed:
Documentation de l'affichage des alertes

Aussi, renommage du paramètre `solve_url` en `alert_solve_url`, par
cohérence.

AmauryCarrade committed:
Correction des liens vers la norme dans la documentation

AmauryCarrade committed:
Typos dans la documentation (merci artragis)

AmauryCarrade committed:
Sémantique dans les galeries

AmauryCarrade committed:
Correction de problèmes de défilement horizontaux

- Les listes de blocs de contenu étaient trop large sur mobile.

AmauryCarrade committed:
Défilements horizontaux : épisode 2, le pied

- Les liens du pied de page étaient trop large sur tablette. Ils ont
  été réduits en reformulant en fonction de la largeur de l'écran.

AmauryCarrade committed:
Défilements horizontaux : épisode 3, le retour des cartes

AmauryCarrade committed:
Typos in documentation (thanks Situphen)

Co-authored-by: Situphen <Situphen@users.noreply.github.com>

AmauryCarrade committed:
La bannière globale masquait le dernier élément du menu

AmauryCarrade committed:
$size-* renommé en $font-size-*

Plus clair et évite des conflits lorsque la norme est utilisée dans
d'autres projets (par exemple avec Bulma).

AmauryCarrade committed:
Mise à jour des normes dans la documentation

AmauryCarrade committed:
Mise à jour des liens dans les normes

AmauryCarrade committed:
Alignement correct des titres de l'accueil.

- Ils sont reconstruits avec flex.
- À bas les flottants.
- Qu'ils coûlent.

AmauryCarrade committed:
Alignement correct des titres avec boutons.

- Ils sont reconstruits avec flex, eux aussi.
- À bas les flottants, encore et encore.
- Qu'ils coûlent, d'accord, d'accord.

AmauryCarrade committed:
Manque de marges entre sections de liste de contenus

AmauryCarrade committed:
Correction de l'alignement des icônes des titres

AmauryCarrade committed:
Manque de marges etre sections de liste de contenus

Pour de vrai

Arnaud-D committed:
Corrige les notifications de ping persistantes sur les réactions à des contenus

AmauryCarrade committed:
“Cette réponse m'a aidé” est réduit pour l'équipe

- Lorsqu'un membre de l'équipe voit un message dans un sujet qu'il n'a
  pas créé, le bouton « Cette réponse m'a aidé » est réduit.

AmauryCarrade committed:
La date est raccourcie sur les petits écrans

- Correction d'un fonctionnement antérieur qui ne marchait plus.

AmauryCarrade committed:
Correction de l'affichage du bouton « m'a aidé »

- Le bouton était déséquilibré lorsqu'affiché sans texte.

AmauryCarrade committed:
Replaced “is author” with a feather icon

- CC BY FontAwesome (TODO add attribution)

AmauryCarrade committed:
Added CSS tooltips and used them for the messages.

- Added CSS tooltips, to use when we need a directly displayed tooltip,
  but don't want to use the JS ones.
- Used these to display “Subject author” and the post edition date,
  on hover, without delay.
- These would be usable for other things as well.

AmauryCarrade committed:
Inclusivité de l'interface des messages

AmauryCarrade committed:
Les infobulles CSS marchent aussi avec data-tooltip

- Si c'est pertinent, on peut utiliser aria-label, mais dans le cas
  contraire, par exemple si l'infobulle est annexe et non descriptive
  de l'ensemble de l'élément, data-toolip peut être utilisé.

AmauryCarrade committed:
Attribution de FA

AmauryCarrade committed:
Adaptation mobile des messages

- Les messages s'affichent correctement sur un téléphone portable ou
  autre petit écran.
- La date et l'info de modification sont déplacées sous le message,
  et la date de modification est dans une modale.
- Sur les très petits écrans, le bouton de citation/modification est
  réduit, afin de laisser la place aux casquettes, si besoin.

Technique

- Ajout d'une media-query `tiny` (et d'une mixin qui va avec) pour les
  très petits écrans.
- Réorganisation du CSS des messages afin que les adaptations mobiles
  soient dans les styles directement, et non à la fin du fichier.

Situphen committed:
Mise à jour de sécurité de Django et Pillow

Situphen committed:
Précise la version de Coveralls à utiliser pour la CI

AmauryCarrade committed:
Amélioration des messages masqués

- Retrait du JS utilisé pour afficher/masquer les messages. À la place,
  l'élément standard `details` est utilisé.
- Amélioration des styles, marges, etc.
- Correction d'un bug de marges si le dernier message d'un vieux sujet
  est masqué.

AmauryCarrade committed:
Amélioration des messages masqués mobile

- Correction de marges
- Correction d'autres marges
- Retrait de marges
- Ajout de marges
- Suppression de bouts d'avatars récalcitrants
- J'ai aussi ajouté une marge

AmauryCarrade committed:
Ajout de la documentation de `horizontal-list`.

AmauryCarrade committed:
Ajout de la documentation de la largeur/mixin `tiny`.

AmauryCarrade committed:
Modification des infobulles CSS et documentation.

- Par défaut, les infobulles CSS utilisent l'attribut `aria-describedby`.
- Il est toujours possible d'utiliser `aria-label` ou `data-tooltip`.
- Ajout de la documentation de ces infobulles.

AmauryCarrade committed:
Correction des tests

- Le test zds.forum.tests.tests_views.MessageActionTest.test_hide ne
  testait plus le bon élément suite aux changements de la façon
  d'afficher le message masqué.
- Le test zds.mp.tests.tests_views.TopicViewTest.test_available_actions
  révélait effectivement un oubli : il était possible demander à masquer
  un message dans un MP.

AmauryCarrade committed:
Le masquage était bien trop masqué

- Certains messages masquables n'avaient pas l'option de disponible.

AmauryCarrade committed:
Ajout du karma sur la version mobile

- Le karma global de l'utilisateur est affiché dans le pied du message,
  sur mobile.

AmauryCarrade committed:
Renommage d'une classe incohérente

- `date-mobile` s'appelle désormais `metadata-mobile`, car il y a bien
  plus que juste la date dans cet élément désormais.

artragis committed:
Amélioration des boutons de connexion via les réseaux sociaux (#5612)

* 🍱 Ajout des icônes de Google

* 💄 Modification du bouton Google+ en bouton Google

* Boutons de connexion avec les bons designs

- Ajout des icônes de Google et Facebook actualisées au format SVG.
- Suppression des sprites de Google+, Google, Foursquare (??).
- Modification des boutons de connexion avec les couleurs et les règles de
  Google et Facebook.

* Adaptation aux normes et correction mobile

- Usage des variables des normes graphiques.
- Correction de l'affichage sur mobile.

* Perfectionnism is key

Co-authored-by: Situphen <Situphen@users.noreply.github.com>

Co-authored-by: Helmasaur <contact@helmasaur.fr>
Co-authored-by: Amaury Carrade <amaury@carrade.eu>
Co-authored-by: Situphen <Situphen@users.noreply.github.com>

AmauryCarrade committed:
Le drapeau de spam potentiel est disponible partout (#6016)

* Le drapeau de spam potentiel est disponible partout

Ce commit déplace le drapeau de spam potentiel (`is-potential-spam`)
vers le modèle parent Comment, ainsi que la vérification des modifica-
tions de messages et le lever d'alerte. L'alerte est levée en respec-
tant le bon `scope` en fonction du type de commentaire.

Une migration est ajoutée pour copier les données depuis le modèle
`forum.Post` vers `utils.Comment`.

* Évite un import (suggestion de Situphen)

Co-authored-by: Situphen <Situphen@users.noreply.github.com>

* Utilise une méthode dédiée pour récupérer la sous-classe de Comment

Co-authored-by: Situphen <Situphen@users.noreply.github.com>

* Lint

* Utilisation de signaux pour envoyer les alertes.

Auparavant, les alertes étaient envoyées depuis la méthode
`update_content` du modèle `utils.Comment`. Le problème, c'est que lors
de l'appel de cette méthode, l'éditeur n'est généralement pas modifié,
ce qui rend son comportement incohérent car l'alerte n'est envoyée que
si l'auteur lui-même modifie le message. Un signal `post_save` est donc
utilisé pour envoyer l'alerte.

* Mise à jour des tests pour prendre tous les cas en compte.

Des tests ont été ajoutés pour les contenus, et les tests du forum ont
été ajustés. De plus, les parties communes des tests forum et contenus
ont été factorisées.

* Removed backup file

* Restaure les pings comme avant

* Vraiment comme avant

* Oh en fait la sauvegarde dans `update_content` était nécessaire

* Forgot to update comment after method behaviour change

Co-authored-by: Situphen <Situphen@users.noreply.github.com>

* Misleading test comment

This survived the previous code review, oups

Co-authored-by: Situphen <Situphen@users.noreply.github.com>

* On n'enregistre plus Comment dans un état incohérent

Ce commit évite d'enregistrer des modèles Comment dans un état
inconsistant dans la base de données. Auparavant, les messages étaient
généralement enregistrés deux fois : une fois avec juste le contenu
mis à jour (mais pas les métadonnées : à ce stade, l'enregistrement est
donc incohérent avec les données réelles), puis une autre avec cette
fois les métadonnées (à ce stade, c'est donc bon).

Le souci, c'est que l'analyse de l'antispam, effectuée automatiquement
à l'enregistrement, a besoin de données cohérentes (à savoir le nouveau
contenu et l'éditeur du message). Cet état de fait faisait donc des
alertes parasites ou manquantes dans certains cas.

Ce commit corrige ce comportement en s'assurant qu'un Comment n'est
jamais enregistré en base de données dans un état incohérent, et en
déplaçant l'analyse des pings et celle de l'anti-spam dans la méthode
`save`.

Au passage, les performances sont très légèrement améliorées, étant donné
qu'un certain nombre de requêtes inutiles sont évitées.

* Apply suggestions from code review

Co-authored-by: Situphen <Situphen@users.noreply.github.com>

* Update misleading documentation

Co-authored-by: Situphen <Situphen@users.noreply.github.com>

AmauryCarrade committed:
Inutile d'afficher le bouton d'antispam quinze fois.

AmauryCarrade committed:
Les signatures s'ouvrent à nouveau au clic.

- Ouverture au clic des signatures qui dépassent une
  ligne.
- Ce système n'utilise plus de javascript.

AmauryCarrade committed:
“Réponse utile” dans le menu si on est modérateur⋅trice.

- Lorsqu'un modérateur ou une modératrice consulte un
  sujet dont iel n'est pas l'auteur, le bouton “cette
  réponse est utile” est déplacé dans le menu de dépas-
  sement. Lorsqu'iel consulte un message dont iel est
  l'auteur⋅trice, par contre, le bouton est toujours
  affiché au même endroit.

AmauryCarrade committed:
Simple infobulle sur la date sur mobile sans droits

- La date de modification est affichée uniquement avec
  une infobulle si on n'a pas les droits d'afficher
  l'historique.

AmauryCarrade commented:
On en est où ici ? :)

Il reste à rebaser et résoudre les conflits, si je ne m'abuse (n'hésite pas à me pinguer si t'as besoin d'aide).

Situphen committed:
Passage de django-cors-middleware à django-cors-headers

AmauryCarrade committed:
Ajout d'un test sur la position de “Réponse utile”

- La position (et la présence) du bouton “Cette réponse
  est utile” est vérifiée grâce à un test.

AmauryCarrade committed:
Pas de case à cocher dans les signatures

- Les signatures s'affichent correctement même sur
  tablette.
- Retrait d'un CSS totalement inutile (lié).

AmauryCarrade committed:
Marges sur tablette du contenu d'un message masqué

AmauryCarrade committed:
Abolition des privilèges des signatures

- On peut afficher les signatures au complet même sans
  être connecté⋅e.

AmauryCarrade committed:
Affichage réduit de “réponse utile” sur petits écrans

AmauryCarrade committed:
Amélioration de l'infobulle d'édition sur mobile

- En tant que simple utilisateur, l'infobulle affiche
  l'éditeur d'un message, lorsqu'il est différent de son
  auteur.

AmauryCarrade committed:
Apply suggestions from code review

Co-authored-by: Situphen <Situphen@users.noreply.github.com>

AmauryCarrade committed:
Code mort

Arnaud-D committed:
Corrige l'affichage des alertes sur les billets (#6011)

Situphen committed:
On peut se déconnecter depuis un mobile

AmauryCarrade committed:
Retour de la coloration syntaxique (oups)

AmauryCarrade committed:
Amélioration du masquage des messages

- Le nombre de messages masqués est affiché, lorsque réduit.

AmauryCarrade committed:
Fixed touch interactions with messages menu (I guess)

- On touch devices, the opening and closing of the menu is handled
  manually; we no longer rely on the “hover-to-touch-event” automatic
  conversion.

AmauryCarrade committed:
Renamed message-accessibility.js for coherence.

AmauryCarrade committed:
Fixed JS error. This being said, the quote autoscroll no longer work.

AmauryCarrade committed:
Le lien de karma d'un membre est correct sur mobile.

- Le lien pointait vers la page courante, et non vers la page de
  profil.

AmauryCarrade committed:
Le bouton “réponse utile” s'inverse au clic dans le menu.

- L'inversion du bouton ne fonctionnait que sous le message, et pas
  dans le menu de modération.

AmauryCarrade committed:
Try to fix double-tap to close menu

Situphen committed:
Les boutons ne s'empilent plus sur les titres des parties d'un contenu (#6035)

AmauryCarrade committed:
N'affiche que les trois points du menu des message sur mobile (#6037)

* N'affiche que les trois points du menu des message sur mobile

* Les points du menu glisse désormais

- La graisse de la police des points du menu a été augmentée, pour
  plus de lisibilité.

Situphen committed:
Adieu node-sass et node-gyp, bienvenue dart-sass ! (#6020)

Situphen committed:
Ressuscite l'icône de Github

Situphen committed:
Mise à jour triviale des dépendances Python

AmauryCarrade committed:
Thème de coloration syntaxique accessible et aux couleurs de ZdS (#6010)

* Essai d'un thème de coloration ZdS

- Inspiré d'a11y-light
- Accessibilité vérifiée

* Adaptation du thème pour CodeMirror

- Suppression du thème IDEA
- Intégration du thème au CSS principal

* Retrait d'une dépendance inutile

- On n'utilise plus highlight.js

* Léger affinement de la colorisation du code

* Le code glisse désormais

- La graisse de la police des codes sources a été augmentée, pour
  plus de lisibilité.

* Amélioration continue de la coloration

* Lignes plus discrètes en console

* Lisibilité du CSS coloré

* Amélioration de la coloration de CodeMirror

- Liens non soulignés
- Textes barrés mais pas leurs délimiteurs
- Titres complètement colorés et plus sombres

* Les nombres sont plus visibles

* Commentaires plus discrets

viki53 committed:
Renomme header-dropdown

viki53 committed:
Composant CSS dropdown avec `details`

viki53 committed:
Utilisation dropdown pour actions message

viki53 committed:
Doc composant dropdown

viki53 committed:
Fix code review

viki53 committed:
Fix SCSS dropdown

Situphen committed:
On enlève un peu de graisse aux blocs de code (#6042)

viki53 committed:
Amélioration dropdown

philippemilink committed:
Ne souligne pas le texte d'un bouton lors du hover (fix #5366) (#5471)

Co-authored-by: Situphen <Situphen@users.noreply.github.com>

artragis committed:
[v30]: fix le bug de l'antispam qui empêchait les spoilers d'être interprétés (#6044)

AmauryCarrade committed:
Les fixtures ne référencent plus les permissions par PK

Les clefs naturelles sont utilisées à la place, pour tout ce qui
concerne les utilisateurs.

Fixes #6022
Fixes #4266

viki53 committed:
Fix longueurs décalage & label dropdown

viki53 committed:
Suppression label répétif

entwanne committed:
Ajout d'un filtre par tag sur les flux RSS/Atom des contenus

entwanne committed:
fixup! Ajout d'un filtre par tag sur les flux RSS/Atom des contenus

entwanne committed:
fixup! Ajout d'un filtre par tag sur les flux RSS/Atom des contenus

entwanne committed:
Strip des valeurs des filtres category/subcategory sur les flux RSS/Atom

entwanne committed:
Relaie les paramTres GET de la page courante aux flux RSS

viki53 committed:
Fix alignement dropdown messages

viki53 committed:
Fix dropdown sur Chrome

viki53 committed:
Tweak ombre+bordure contenu dropdown

viki53 committed:
Ajout padding dans dropdown messages

viki53 committed:
Retire hauteur fixe éléments menu messages (#6052)

Situphen committed:
Ajoute une commande pour vider le cache (#6049)

Co-authored-by: Amaury Carrade <amaury@carrade.eu>

Situphen committed:
Amélioration de la lisibilité avec OpenDyslexic

viki53 committed:
Éditeur: Barre d'outil sticky sur tablette+ (#6051)

* Éditeur: Barre d'outil sticky sur tablette+

* Décalage en cas de bannière

Co-authored-by: Amaury Carrade <amaury@carrade.eu>

Situphen committed:
Suppression d'une marge inutile

viki53 committed:
Fix hauteur body et z-index header

Situphen committed:
Correction de l'affichage des membres sur les pages des casquettes

Situphen committed:
Survoler le sous-titre d'une cartouche de sujet du forum ne souligne plus le titre

Situphen committed:
On cache le bouton Modifier dans les MPs si quelqu'un a répondu après

Situphen committed:
Un membre ne peut pas modifier son message s'il est masqué

Situphen committed:
Rend moins visible un paragraphe

Situphen committed:
Cache les boutons Spam potentiel et Réponse utile pour les messages masqués

Rowin committed:
Change l'URL de la page de profil en /@pseudo (#6057)

* Fix 6047
Nouvelle url plus simple pour le profil utilisateur
Redirection de l'ancienne url

* Ajout des tests

* Mise à jour du test vérifiant l'url du profil

* Enlever une ligne (commentée) de l'ancien code

Co-authored-by: Situphen <Situphen@users.noreply.github.com>

* Changement du template pour afficher la bonne url

* Changement de l'URL du profile inscrite en dur

Co-authored-by: Situphen <Situphen@users.noreply.github.com>

Situphen committed:
Passage à zmarkdown 9.1.4 pour corriger un bug

Situphen committed:
Oh la belle typo !

Situphen committed:
Déplace SubCategoryFactory et LicenseFactory dans le module utils

Rendre à César ce qui lui appartient

Situphen committed:
Passage à zmd 9.1.5 pour corriger les notes de bas de page (#6071)

Situphen committed:
Met à jour Pillow en 8.1.2 (#6072)

viki53 committed:
Ajoute espace avant alerte dans formulaire de réponse (#6062)

Situphen committed:
Ajout d'un precommit-hook pour pyupgrade

Situphen committed:
Petite cure de jeunesse du code

Migwel committed:
Forcer la limite de caractères pour le texte de masquage #6017 (#6076)

Migwel committed:
Correction du lien vers Github

artragis committed:
Plus de problèmes lors des changmenets d'heure dans les rss

artragis committed:
Enlève la duplication d'id

artragis committed:
change les id pour ask validation et cancel validation

artragis committed:
change les id pour refus de validation

artragis committed:
fix test

Arnaud-D committed:
Généralise l'usage de force_login dans les tests quand approprié (#6074)

* Utilise force_login dans tests_editcontentlicense.py

* Utilise force_login dans tests_utils.py

* Utilise force_login dans tests_opinion_views.py

* Utilise force_login dans tests_move.py

* Utilise force_login dans tests_models.py

* Utilise force_login dans tests_lists.py

* Utilise force_login dans zds/features/tests.py

* Utilise force_login dans zds/forum/api/tests.py

* Utilise force_login dans zds/forum/tests/tests.py (+ petit fix)

* Utilise force_login dans zds/forum/tests/tests_feeds.py

* Utilise force_login dans zds/forum/tests/tests_views.py

* Utilise force_login dans zds/gallery/tests/tests_views.py

* Utilise force_login dans zds/tutorialv2/tests/tests_views/tests_content.py

* Utilise force_login dans zds/tutorialv2/tests/tests_views/tests_edgecases.py

* Utilise force_login dans zds/tutorialv2/tests/tests_views/tests_editcontenttags.py

* Utilise force_login dans zds/tutorialv2/tests/tests_views/tests_published.py

* Utilise force_login dans zds/tutorialv2/tests/tests_views/tests_stats.py

* Utilise force_login dans zds/utils/tests/tests_comments.py

* Utilise force_login dans zds/utils/tests/tests_interventions.py

* Utilise force_login dans zds/utils/managegement

* Utilise force_login dans zds/tutorialv2/api

* Utilise force_login dans zds/searchv2/tests

* Utilise force_login dans zds/pages/tests.py

* Utilise force_login dans zds/notification/tests/tests_tricky.py

* Utilise force_login dans zds/notification/tests/tests_basics.py

* Utilise force_login dans zds/mp/tests/tests_utils.py

* Utilise force_login dans zds/mp/tests/tests_views.py

* Utilise force_login dans tests_setlastvisitmiddleware.py

* Utilise force_login dans zds/member/tests/tests_views.py

philippemilink committed:
Ajoute un filtre sur l'état de lecture des notifications dans l'API

Fix #6073

Migwel committed:
Interdire l'utilisation de / dans le pseudo #5984 (#6079)

* Interdire l'utilisation de / dans le pseudo #5984

* Améliore le message d'erreur + renomme le test

artragis committed:
Permet l'utilisation de svg dans la gallerie (#6067)

Arnaud-D committed:
Corrige un bug dans l'abonnement aux MP (#6070)

* Corrige un bug dans l'abonnement aux MP
Utilise add_participant dans tous les cas d'ajout de participant à un MP (sauf pour l'API).
Ajoute une option au signal pour distinguer les usages explicites (qui doivent conduire à une notif)
de ceux silencieux (qui ne doivent pas conduire à une notif).

* Ajoute un signal pour la création d'un MP
Ce signal permet de découpler la gestion des abonnements à la création du topic et la gestion des notifications en cas de nouveaux messages dans le topic.

* Corrige le receiver pour topic_created

* Faute de frappe dans un commentaire

* Corrige un problème de notifications par mail

* Corrige le test de l'API

* Refactorise les MP pour simplifier la fonction utilitaire

* Fix nom de variable

* Fix arguments in definition of signals

* Fix management of exception NotReachable error in MP function

Migwel committed:
Envoie un MP lorsqu'un membre est retiré des auteurs d'un contenu (#5778) (#6085)

Situphen committed:
Relooking des pages d'erreurs statiques (#6098)

Situphen committed:
Mise à jour des dépendances Python (#6099)

* Mise à jour de sécurité des dépendances Python

* Mise à jour triviale des dépendances Python

Migwel committed:
Lier des sujets du forum à des dépôts annexes #5807 (#6084)

* Suppression des tags lors de la création d'un ticket Github

* Lier des sujets du forum à des dépôts annexes

* Fix lint-back

* Améliore la lisibilité du code

* Rendre le choix de dépôt plus flexible

* Permet d'associer des sujets à des tickets existant dans des dépôts annexes

* Simplification du code

artragis committed:
Réduit la quantité de requêtes réalisées sur la page d'accueil des forums (#6089)

* Enlève des query en double dans la page d'accueil du forum (1/2)

- retire la double sélection du first_post
- met un select_related sur le topic pour éviter une nouvelle requête
- 25% d'amélioration en dev pour les anonymes

* Réduction du nombre de requêtes sur le forum 2/2

- pour les utilisateurs connectés
- 12% de réduction du temps de processing en dev
- quelques ajouts dynamiques de propriétés ajoutées pour faciliter la vie

* Fin des optimisations

J'ai trouvé un truc assez énorme quand on fait pas gaffe avec les select_related:
Quand on select une référence vers l'objet lui-même ça empêche
l'edition par la suite
e.g : sélectionner le first_message d'un topic en faisant
un select_related("topic") alors qu'on est en train
d'éditer le topic.

* Opti pour les requêtes issus de back-ref dans les templates

* enlève un block inutile

Co-authored-by: Situphen <Situphen@users.noreply.github.com>

Situphen committed:
Optimisation des requêtes SQL sur la page d'accueil

Situphen committed:
Prise en compte des remarques de artragis

artragis committed:
Change la manière de faire la publication des tuto et des latex (#6025)

* Change la manière de faire la publication des tuto et des latex

* Reformate et améliore le code pour le rendre plus robuste

* Premier fix de zmd pour les niveaux de titre et retrait du titre dans le latex

* Change la manière de faire la publication des tuto et des latex

* Reformate et améliore le code pour le rendre plus robuste

* Publication par manifest

* reformatte

* change zmd

* fix md publicator bug

* fix md publicator bug

* fix zmd expected result with <a> tabindex

* Black

* Fix html format issue

* FIx SYNC

* zmd à la dernière version

* debug test

* debug test

* debug test

* debug test

* debug test

* remove useless log info

* formatter

* améliore la documentation

* Fix le bug de la publication markdown & des stats

* Republish markdown

* Use right order for html

* Fix markdown generation

Co-authored-by: Situphen <Situphen@users.noreply.github.com>

philippemilink committed:
Supprime le paramètre enable_pings devenu inutile

Situphen committed:
Remplace le badge Travis CI par celui de Github Actions

philippemilink committed:
Supprime des fichiers qui n'ont rien à faire dans le dépôt

Arnaud-D commented:
@ToxicScorpius Est-ce que tu sais remettre en ordre l'historique des commits sur cette PR ? Sinon, est-ce que tu peux lister les commits qui contiennent ton code ? J'ai essayé de les identifier pour remettre tout en ordre, mais il y en a au début et à la fin de la liste, j'ai du mal à voir qui est qui et à bien prendre toutes tes modifications.

Arnaud-D commented:
Salut, j'ai de nouveau jeté un œil, et c'est difficile de remettre au propre à ta place ou sans toi en direct. Après quelques essais, le plus simple est probablement de faire de la transplantation. Autrement dit, de mettre de côté tes modifications dans un état qui te satisfait (copier les fichiers modifiés quelque part par exemple), et de recréer tes commits à la main sur une base saine.

On peut faire autrement avec du cherry-picking et autre, mais c'est laborieux (en tout cas trop pour moi).

SpaceFox commented:
Je pense voir ce qu’il y a à faire pour récupérer cette branche – effectivement comme le dit @Arnaud-D le mieux serait de refaire les modifications sur une base saine, les commits contiennent trop d’erreurs pour être simplement _cherry-pick_, en particulier ceux qui contiennent des caractères de gestion de conflit (`<<<<<<< HEAD` et consorts).

@ToxicScorpius tu veux t’en charger ou je reprends ton travail ?

Arnaud-D commented:
Reprise par #6401.
